prevent screen from juddering when clicking on diagram

Maschga opened the pull request:
Hi,
If you click on the `In-Out-diagram`, the screen jerks:

https://github.com/user-attachments/assets/79b26735-fac4-4840-b1b8-0f3ac452483a

Adding `overflow-y: scroll` to the `<body>`-tag prevents this:

https://github.com/user-attachments/assets/0de8fcf0-d003-4631-83d2-d0f87c15b016

Maschga committed:
add `overflow-y: scroll`

naltatis commented:
We didnt to this to avoid double scrollbar issues with scrollable modals. See loadpoint settings in demo config:

![Bildschirmfoto 2024-09-02 um 11 52 13](https://github.com/user-attachments/assets/f8821393-dca7-466e-9f28-fe9701db814e)

But I guess doing this change except when modals are open would be fine. Can you test this on your devices?

```
body:not(.modal-open) {
	overflow-y: scroll;
}
```

Maschga commented:
Hi, I use Firefox and I do not have this problem with the doubled scrollbar:
https://github.com/user-attachments/assets/fb72c454-87b1-4b07-9115-afd50bd2c76f

However, I can move the screen slightly to the right. I think this is not intended? Do you have the same issue?

naltatis commented:
> Do you have the same issue?

No i dont. Is your video with or without the overflow change? Firefox on Android?

Maschga commented:
I have this behaviour with and without the overflow-change.

Maschga commented:
It's due to the footer:
https://github.com/user-attachments/assets/c83f6563-c93a-4ba2-ba50-7a027219510c

naltatis commented:
> It's due to the footer

What device/resolution? We already have a "short version" for the "100% solar" text but it doesn't seem to be short enough. at lease for dev-builds with long version.
Can you try to add text truncation for the text. Not sure if it's easy in combination with the flex design.
Other languages likely have even longer words.

Maschga commented:
Samsung Galaxy A71 with 1080 x 2400 px.

Before the text is cut off, I suggest using mediaqueries to place it in two lines in the centre if the screen is too small or the font size too large.
I can try it out tomorrow, I would like to do it.

naltatis commented:
Here are all translations for this text:
https://hosted.weblate.org/search/evcc/evcc/?q=%5Bfooter.savings%5DfooterShort&sort_by=-priority%2Cposition&checksum=

Maschga committed:
add `overflow-y` if no modal is open

Maschga committed:
using mediaqueries to wrap text in two lines

Maschga commented:
I added this:
```css
body:not(.modal-open) {
	overflow-y: scroll;
}
```
and added a media query, which wraps the text in two lines if the screen is too small.

On my devices (computer + smartphone) both issues are solved.

Does this work on your devices, too?
Does this solution suit you?

naltatis committed:
truncate version number instead of wrap

naltatis commented:
Scrolling tests look fine. Let's give it a try.

I replace the wrap with truncate of the version number. This is the least important information in the footer.

Generally I also like the two line wrap but we should not do it based on a fixed media query. A proper solution would be content based because translation texts may vary in width. Flex wrap does this by default, but i dont see a proper way to do the centering in case of a wrap. So lets keep it simple and just truncate/clip the version part when space is low.

<img width="365" alt="Bildschirmfoto 2024-09-05 um 22 23 15" src="https://github.com/user-attachments/assets/dd888b36-6ddf-482d-b459-8a553631106b">
